homeassistant/components/met/__init__.py

```diff
@@ -56,19 +59,20 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+async def async_unload_entry(
+    hass: HomeAssistant, config_entry: MetWeatherConfigEntry
+) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(
         config_entry, PLATFORMS
     )
 
-    hass.data[DOMAIN][config_entry.entry_id].untrack_home()
-    hass.data[DOMAIN].pop(config_entry.entry_id)
+    config_entry.runtime_data.untrack_home()
```

Comment:
This could be replaced with `config_entry.async_on_unload(coordinator.untrack_home)` in `async_setup_entry`.